### PR TITLE
feat(gitops): Add ArgoCD support

### PR DESCRIPTION
**Pull request by KasnocknDave**
This PR adds ArgoCD support and extends the readme.

### CODE REVIEW

**Review by wolfmayr — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -65,21 +65,28 @@ Execute setupProvider.sh to install the Github-provider and its access credentia
...
+Execute setupArgoCD.sh to install ArgoCD and set up the apps.
 
-`cd ..`
+`./setupArgoCD.sh`
 
-Apply the XRDs.
+Finally, execute connectArgoCD.sh to prompt the local admin password and port forward ArgoCD to localhost:8080.
 
-`kubectl apply -f xrds/contribution/`
+`./connectArgoCD`
```
Comment (oskisk) [inline README.md]:
```suggestion
`./connectArgoCD.sh`
```

File: README.md [hunk 2]
```diff
@@ -65,21 +65,28 @@ Execute setupProvider.sh to install the Github-provider and its access credentia
...
 
-Finally, apply the claim.
+### Check and sync the resources in ArgoCD
 
-`kubectl apply -f claims/customRepo.yaml`
+Call `localhost:8080` and use the following credentials for the initial login:
```
Comment (oskisk) [inline README.md]: Would be nice to make this into a link to click.

**Review by Bleggbeard — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -65,21 +65,28 @@ Execute setupProvider.sh to install the Github-provider and its access credentia
...
-Finally, apply the claim.
+### Check and sync the resources in ArgoCD
 
-`kubectl apply -f claims/customRepo.yaml`
+Call [http://localhost:8080](http://localhost:8080) and use the following credentials for the initial login:  
 
-### Check the resources
+```
+user: admin  
+password: the password prompted by the connectArgoCD.sh script
```
Comment (Bleggbeard) [inline README.md:82]:
```suggestion
password: (the password prompted by the connectArgoCD.sh script)
```

File: setup-scripts/connectArgoCD.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+echo "Wait for ArgoCD to be up and healthy."
+kubectl wait deployment -n argocd argocd-server --for condition=Available=True --timeout=90s
+echo "Initial admin password: $(kubectl get secret -n argocd argocd-initial-admin-secret -o yaml | grep "password:" | sed 's/  password: //' | base64 -d)"
+kubectl -n argocd port-forward service/argocd-server 8080:80
```
Comment (Bleggbeard) [inline setup-scripts/connectArgoCD.sh:7]: You can add a 
`echo "ArgoCD URL: http://localhost:8080"`
or something similar.

**Review by oskisk — APPROVED** [review]

**Pull request merged**